daemon/call_interfaces.c

```diff
@@ -2102,6 +2102,35 @@ found:;
 #endif
 }
 
+void call_interfaces_free() {
+	if (info_re) {
+		pcre_free(info_re);
+		info_re = NULL;
+	}
+
+	if (streams_re) {
+		pcre_free(streams_re);
+		streams_re= NULL;
+	}
+
+	if (info_ree) {
+#ifdef PCRE_CONFIG_JIT
```

Comment:
Curious about this one - I've used `pcre_free_study` unconditionally elsewhere in the code. Is there something that I've missed?

Reply:
Deleting the whole "if" for "info_ree", valgrind reports the following on my machine:
```
==17304== 108 bytes in 1 blocks are still reachable in loss record 238 of 267
==17304==    at 0x4C2FECB: malloc (vg_replace_malloc.c:307)
==17304==    by 0x576E64B: pcre_study (in /lib/x86_64-linux-gnu/libpcre.so.3.13.3)
==17304==    by 0x1484C2: call_interfaces_init (call_interfaces.strhash.c:2378)
==17304==    by 0x119CB7: init_everything (main.c:758)
==17304==    by 0x119CB7: main (main.c:902)
```

Reply:
Sure, you need to free it, but I'm wondering about the PCRE_CONFIG_JIT define. I thought it would be enough to just call `pcre_free_study` without having to make a case distinction?

Reply:
Ah, when I was searching of how to do the pcre free for study I found an example [1]:
```
...
so that no option is passed to pcre_study(), and then use something like this to free the study data:

  #ifdef PCRE_CONFIG_JIT
      pcre_free_study(study_ptr);
  #else
      pcre_free(study_ptr);
  #endif
```
... and I just did it like there.

[1] https://www.pcre.org/original/doc/html/pcrejit.html#SEC4

Reply:
However, trying now ```pcre_free_study```/```pcre_free``` separately, without any #ifdef conditionals, both work ok.

So I think you are right, there's no real need for that #ifdef conditional in the current rtpengine code and I can use just ```pcre_free_study``` for this.